src/main/scala/com/amazon/deequ/analyzers/catalyst/StatefulCorrelation.scala

```diff
@@ -50,4 +50,8 @@ private[sql] class StatefulCorrelation(
     val state = Seq(super.hashCode(), evaluateExpression)
     state.map { _.hashCode() }.foldLeft(0) {(a, b) => 31 * a + b }
   }
+
+  override protected def withNewChildrenInternal(newLeft: Expression,
+                                                 newRight: Expression): StatefulCorrelation =
+    new StatefulCorrelation(newLeft, newRight, nullOnDivideByZero)
```

Comment:
This is to fix a fail on `baseCheck.hasCorrelation`
https://github.com/tanvn/deequ/blob/master/src/test/scala/com/amazon/deequ/checks/CheckTest.scala#L577

If we do not provide this `withNewChildrenInternal`, the [method](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Corr.scala#L131-L133) of Corr will be used, which lead to incorrect results.